handlers/handler-fault-injection/src/main/java/org/apache/servicecomb/faultinjection/FaultInjectionUtil.java

```diff
@@ -0,0 +1,168 @@
+   * @return value of config parameter map
+   */
+  public static Map<String, AtomicInteger> getConfigCenterMap() {
+    return configCenterValue;
+  }
+
+  /**
+   * Sets the value for given config parameter.
+   * @param key
+   * @param value
+   */
+  public static void setConfigCenterValue(String key, AtomicInteger value) {
+    configCenterValue.put(key, value);
+  }
+
+  /**
+   * Handles the reading fault injection configuration.
+   * 
+   * @param invocation
+   *            invocation of request
+   * @param key
+   *            configuration key
+   * @return configuration value
+   */
+  public static int getFaultInjectionConfig(Invocation invocation, String key) {
+    int value = 0;
+    String config;
+
+    // get the config base on priority. operationName-->schema-->service-->global
+    String operationName = invocation.getOperationName();
+    String schema = invocation.getSchemaId();
+    String serviceName = invocation.getMicroserviceName();
+
+    config = CONSUMER_FAULTINJECTION + serviceName + ".schemas." + schema + ".operations." + operationName + "."
+        + CONSUMER_FAULTINJECTION_POLICY_PROTOCOLS + invocation.getTransport().getName() + "." + key;
+
+    value = getConfigValue(config);
+    if ((value != FAULT_INJECTION_DEFAULT_VALUE)) {
+      return value;
+    }
+
+    config = CONSUMER_FAULTINJECTION + serviceName + ".schemas." + schema + "."
+        + CONSUMER_FAULTINJECTION_POLICY_PROTOCOLS + invocation.getTransport().getName() + "." + key;
+
+    value = getConfigValue(config);
+    if ((value != FAULT_INJECTION_DEFAULT_VALUE)) {
+      return value;
+    }
+
+    config = CONSUMER_FAULTINJECTION + serviceName + "." + CONSUMER_FAULTINJECTION_POLICY_PROTOCOLS
+        + invocation.getTransport().getName() + "." + key;
+    value = getConfigValue(config);
+    if ((value != FAULT_INJECTION_DEFAULT_VALUE)) {
+      return value;
+    }
+
+    config = CONSUMER_FAULTINJECTION_GLOBAL + CONSUMER_FAULTINJECTION_POLICY_PROTOCOLS
+        + invocation.getTransport().getName() + "." + key;
+
+    value = getConfigValue(config);
+    return value;
+  }
+
+  /**
+   * Get the configuration value
+   * @param config config parameter
+   * @return int value
+   */
+  private static int getConfigValue(String config) {
+    int value = 0;
+    //first need to check in config center map which has high priority.
+    Map<String, AtomicInteger> cfgMap = FaultInjectionUtil.getConfigCenterMap();
+
+    if (cfgMap.containsKey(config)) {
+      return cfgMap.get(config).get();
+    }
+
+    value = FaultInjectionConfig.getConfigVal(config, FAULT_INJECTION_DEFAULT_VALUE);
+    return value;
+
+  }
+
+  /**
+   * It will check the delay/abort condition based on request count and percentage
+   * received.
+   * 
+   * @param reqCount
+   * @param percentage
+   * @param key
+   * @return true/false
+   */
+  public static boolean checkFaultInjectionDelayAndAbort(long reqCount, int percentage) {
```

Comment:
"check"?
return true means need or no need?
method name is not so good.

Reply:
changed method name to "isFaultNeedToInject" and written javadoc for true/false case